docs/plugins/feature-flags.md

```diff
@@ -0,0 +1,64 @@
+---
+id: feature-flags
+title: Feature Flags
+description: Details the process of defining setting and reading a plugin feature flag.
+---
+
+Backstage offers the ability to define feature flags inside a plugin. This allows you to restrict parts of your plugin to those individual users who have toggled the feature flag to on.
+
+This page describes the process of defining setting and reading a plugin feature flag.
+
+## Defining a Feature Flag
+
+Before using a feature flag we must first define it. This is done when we create the plugin by passing the name of the feature flag into the `featureFlags` array.
+
+```ts
+/* src/plugin.ts */
+import { createPlugin, createRouteRef } from '@backstage/core-plugin-api';
+import ExampleComponent from './components/ExampleComponent';
+
+export const examplePlugin = createPlugin({
+  id: 'example',
+  routes: {
+    root: rootRouteRef
+  },
+  featureFlags: [{ name: 'show-example-feature' }],
+});
+
+```
+
+## Enabling Feature Flags
+
+Feature flags are defaulted to off and can be updated by individual users in the backstage interface.   
+
+These are set by navigating to the page under `Settings` > `Feature Flags`. 
+
+The users selection is saved in the users browsers local storage. Once toggled it may be required for a user to refresh the page to see any new changes.
+
+## FeatureFlagged Component
+
+The easiest way to control content based on the state of a feature flag is to use the [FeatureFlagged](https://backstage.io/docs/reference/core-app-api.featureflagged) component.
+
+```ts
+import { FeatureFlagged } from '@backstage/core-app-api'
+
+...
+
+<FeatureFlagged with="show-example-feature">
+    <SampleComponent />
```

Comment:
Could this be `<NewFeatureComponent />` and then on for the "without" be `<PreviousFeatureComponent />`, just to better illustrate the difference?

Reply:
Good idea